docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,791 @@
+
+```csharp
+// MvvmLight
+this.Set<MyObject>(() => this.MyProperty, ref this.myProperty, value, true);
+
+// Toolkit.Mvvm
+this.SetProperty<MyObject>(ref this.myProperty, value, true, nameof(this.MyProperty));
+```
+
+Note, string parameter is not required if the method is being called from the property's setter as it is inferred from the caller member name.
+
+#### Broadcast<T> ( T, T, string )
+
+`Broadcast<T>(T, T, string)` has a direct replacement which doesn't require a rename. 
+
+```csharp
+// MvvmLight
+this.Broadcast<MyObject>(this.oldValue, this.newValue, nameof(this.MyProperty));
+
+// Toolkit.Mvvm
+this.Broadcast<MyObject>(this.oldValue, this.newValue, nameof(this.MyProperty));
+```
+
+Note, the message sent via the `Messenger` property when calling the `Broadcast` method has a direct replacement for `PropertyChangedMessage` within the Toolkit's framework. 
+
+Also, string parameter is not required if the method is being called from the property's setter as it is inferred from the caller member name.
+
+#### RaisePropertyChanged<T> ( string, T, T, bool )
+
+There is no direct replacement for the `RaisePropertyChanged<T>(string, T, T, bool)` method and any code using this should be altered or removed.
+
+```csharp
+// MvvmLight
+this.RaisePropertyChanged<MyObject>(nameof(this.MyProperty), this.oldValue, this.newValue, true);
+
+// Toolkit.Mvvm
+// No direct replacement, remove
+```
+
+#### RaisePropertyChanged<T> ( Expression, T, T, bool )
+
+There is no direct replacement for the `RaisePropertyChanged<T>(Expression, T, T, bool)` method.
+
+It is recommended for improved performance that you replace this with the Toolkit's `RaisePropertyChanged<T>(string, T, T bool)` using the `nameof` keyword instead.
+
+```csharp
+// MvvmLight
+this.RaisePropertyChanged<MyObject>(() => this.MyProperty, this.oldValue, this.newValue, true);
+
+// Toolkit.Mvvm
+this.RaisePropertyChanged<MyObject>(nameof(this.MyProperty), this.oldValue, this.newValue, true);
+```
+
+### ViewModelBase Properties
+
+#### MessengerInstance
+
+`MessengerInstance` has a renamed direct replacement, `Messenger`.
+
+```csharp
+// MvvmLight
+IMessenger messenger = this.MessengerInstance;
+
+// Toolkit.Mvvm
+IMessenger messenger = this.Messenger;
+```
+
+#### IsInDesignMode
+
+There is no direct replacement for the `IsInDesignMode` property and any code using this should be altered or removed.
+
+```csharp
+// MvvmLight
+var isInDesignMode = this.IsInDesignMode;
+
+// Toolkit.Mvvm
+// No direct replacement, remove
+```
+
+### ViewModelBase Static Properties
+
+#### IsInDesignModeStatic
+
+There is no direct replacement for the `IsInDesignModeStatic` property and any code using this should be altered or removed.
+
+```csharp
+// MvvmLight
+var isInDesignMode = ViewModelBase.IsInDesignModeStatic;
+
+// Toolkit.Mvvm
+// No direct replacement, remove
+```
+
+## Migrating RelayCommand
```

Comment:
Not strictly retaled, but we might also want to briefly touch upon the `AsyncRelayCommand` type here?
As in, I would imagine quite a few developers to have been using the standard `RelayCommand` from MVVMLight to wrap asynchronous methods instead (thought not directly supported), with workarounds to make that work such as:
```csharp
var command = new RelayCommand(() => MyAsyncMethod());

// Or even worse
var command = new RelayCommand(async () => await MyAsyncMethod());
```

We should mention that if they have anything like that in their code, they can (and should) replace that with the new `AsyncRelayCommand` and `AsyncRelayCommand<T>` types, by changing the above to just:
```csharp
var asyncCommand = new AsyncRelayCommand(MyAsyncMethod);

// Or if it has a parameter
var asyncCommand = new AsyncRelayCommand<ParamType>(MyAsyncMethodWithParam);
```

I fear if we don't mention this new functionality here many devs might just miss it completely and keep using the previous workaround, which is not ideal for a number of reasons 😊

Reply:
I've popped in some separate headered sections in here to call out these specific scenarios so they don't get lost. Good shout though.